### DISCUSSION TITLE
Uno.Wasm AOT performance .net 5

### DISCUSSION THREAD

**Discussion by andrey-shavelev** [unoplatform/uno#9766] [Q&A]
Hello Everyone! What could be the general expectancy for WASM performance improvement after AOT? 

I have wasm project that I managed to AOT-compile recently. It is quite a big project with the Skia used extensively. And it relies on a .net5 runtime. The AOT build takes anywhere from 1 to 2 hours on a decent BA. Surprisingly, after AOT build, I received zero performance improvement. At least, it is not something I could measure. That is a challenging puzzle, but maybe I'm missing something? 
I have to exclude the entry point and its "root" assemblies from AOT because of runtime exceptions, so maybe this affects the overall result?

**Comment by jeromelaban** [comment]
You can measure the runtime performance using [the profiling setting](https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/features-profiling.html#cpu-profiling) and the browser's performance tab. Note that depending on what your code is doing, you may be hitting scenarios that may have been addressed in the bootstrapper 4.0-dev builds. Otherwise, it may be something to raise to the .NET runtime team with a repro case.
